draft-ietf-mmusic-dtls-sdp.xml

```diff
@@ -368,10 +386,12 @@
             <t>
                 If the offerer inserts the SDP 'setup' attribute with an 'actpass' or 'passive' attribute value, the
                 offerer MUST be prepared to receive a DTLS ClientHello message (if a new DTLS association
-                is established by the answerer) from the answerer before the offerer receives the SDP answer.
+                is established by the answerer) from the answerer before the offerer receives the SDP answer. In such
+                case the offerer MAY cache the ClientHello message, but the offerer MUST NOT initiate a DTLS

```

Comment:
I think the main issue here is if we want to allow unverified media. If unverified media is allowed it makes sense to figure out the way to implement early DTLS handshake. If unverified media is treated as a security risk that it is, then caching ClientHello and only proceeding with DTLS handshake when answer or specifically fingerprints and tls-id are received by the offerer is a safer option. I think whatever extensions are developed to make connection setup faster, they should include provisions for fingerprint and tls-id delivery.  Since ufrag and ICE password are delivered to the offerer at this point, nothing should prevent fingerprint and tls-id delivery at the same time.